### PR TITLE
Added programs

### PR DESCRIPTION
**Pull request by saketrule**
Added programs for Segment tree, prime number checker, sieve of eratosthenes, and GCD to get started.
Updated README.md
I checked the program for correctness, but you can test them anyways.

### CODE REVIEW

File: tree/Segment Tree/segment_tree.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+'''
```
Comment (keon) [inline tree/Segment Tree/segment_tree.py]: Le'ts use folder names without blank lines.
i'd prefer just snake case (same as variables)

**Comment by saketrule** [comment]
Done

File: README.md [hunk 2]
```diff
@@ -1,8 +1,12 @@
-# Pythonic Data Structures and Algorithms
```
Comment (outlawmose) [inline README.md]: Thank you

**Comment by outlawmose** [comment]
😉😉

**Comment by keon** [comment]
That was quick :)

**Comment by outlawmose** [comment]
Not quick enough!

**Comment by outlawmose** [comment]
😆

**Pull request merged**